internal/reconcile/uninstall.go

```diff
@@ -138,10 +138,7 @@ func (r *Uninstall) Reconcile(ctx context.Context, req *Request) error {
 	// Handle any error.
 	if err != nil {
 		r.failure(req, logBuf, err)
-		if req.Object.Status.History.Latest().Digest == cur.Digest {
```

Comment:
@cwrau the tests are failing, please run `make test` before opening a PR.

Reply:
Ok, I thought I didn't understand the code well enough 😅

I don't understand why one wouldn't want to return an error in case of an error, even in this test case. The error is `timed out waiting for the condition`, why not return an error to be retried?

Or rather, why is this working in this case, but not in the case of #1021?